src/rebar3_ex_doc.erl

```diff
@@ -554,3 +561,34 @@ mermaid_add(Vsn) when is_list(Vsn) ->
     }}];
 mermaid_add(Vsn) ->
     ?RAISE({mermaid_vsn_not_string, Vsn}).
+
+merge_before_closing_body_tag(MermaidMap, Custom) ->
+    try
+        do_merge_before_closing_body_tag(MermaidMap, Custom)
+    catch
+        _:_:_ ->
+            ?RAISE({invalid_before_closing_body_tag, Custom})
+    end.
+
+do_merge_before_closing_body_tag(undefined, Custom) ->
+    Custom;
+do_merge_before_closing_body_tag(MermaidMap, undefined) ->
+    MermaidMap;
+do_merge_before_closing_body_tag(MermaidMap, {M,F,Args}) 
+        when is_atom(M) and is_atom(F) and is_list(Args) ->
+    CustomMap = #{html => erlang:apply(M, F, [html|Args]),
+                  epub => erlang:apply(M, F, [epub|Args])},
+    do_merge_before_closing_body_tag(MermaidMap, CustomMap);
+do_merge_before_closing_body_tag(MermaidMap, CustomFun)
+        when is_function(CustomFun, 1) ->
+    CustomMap = #{html => CustomFun(html),
+                  epub => CustomFun(epub)},
+    do_merge_before_closing_body_tag(MermaidMap, CustomMap);
+do_merge_before_closing_body_tag(MermaidMap, CustomMap) ->
+    maps:merge_with(fun (html,Mermaid,Custom) ->
+                        Mermaid ++ Custom;
```

Comment:
This will currently fail because a string (such as the script to include mermaid) can be concatenated with an mfa.

Reply:
Is it legal to write something like:
```erlang
{before_closing_body_tag, #{html => fun mod:do_html/1}}
```?
I thought that you can only pass a string literal in a map

Reply:
You are totally right. My bad, thank you 😄